packages/core/TextSearch/TextSearchManager.ts

```diff
@@ -21,148 +21,119 @@ export interface SearchScope {
-    ) {
-      // determine list of relevant adapters based on scope
-      this.textSearchAdapters = this.loadTextSearchAdapters(searchScope)
-      const results: Array<BaseResult[]> = await Promise.all(
-        this.textSearchAdapters.map(async adapter => {
-          // search with given search args
-          const currentResults: BaseResult[] = await adapter.searchIndex(args)
-          return currentResults
-        }),
-      )
+  getAdaptersWithAssembly(
+    searchScopeAssemblyName: string,
+    adapterList: AnyConfigurationModel[],
+  ) {
+    return adapterList.filter(adapterConfig =>
+      readConfObject(adapterConfig, 'assemblies')?.includes(
+        searchScopeAssemblyName,
+      ),
+    )
+  }
 
-      // aggregate and return relevant results
-      const relevantResults = this.sortResults(
-        results.flat(),
-        rankSearchResults,
-      )
+  /**
+   * Returns list of relevant results given a search query and options
+   * @param args - search options/arguments include: search query
+   * limit of results to return, searchType...prefix | full | exact", etc.
+   */
+  async search(
+    args: BaseArgs,
+    searchScope: SearchScope,
+    rankFn: (results: BaseResult[]) => BaseResult[],
+  ) {
+    // determine list of relevant adapters based on scope
+    this.textSearchAdapters = this.loadTextSearchAdapters(searchScope)
+    const results = await Promise.all(
+      this.textSearchAdapters.map(adapter => adapter.searchIndex(args)),
+    )
 
-      if (args.limit && relevantResults.length > 0) {
-        return relevantResults.slice(0, args.limit)
-      }
-      return relevantResults
-    }
+    // aggregate and return relevant results
+    return this.sortResults(results.flat(), rankFn)
+  }
 
-    /**
-     * Returns array of revelevant and sorted results
-     * @param results - array of results from all text search adapters
-     * @param rankSearchResults - function that updates results scores
-     * based on more relevance
-     */
-    sortResults(
-      results: BaseResult[],
-      rankSearchResults: (results: BaseResult[]) => BaseResult[],
-    ) {
-      // first sort results in alphabetical order
-      const sortedResults = results.sort(
-        (a, b) => -b.getLabel().localeCompare(a.getLabel()),
-      )
-      // sort results based on score
-      const sortedScoredResults = rankSearchResults(sortedResults).sort(
-        function (result1: BaseResult, result2: BaseResult) {
-          if (result1.getScore() < result2.getScore()) {
-            return 1
-          }
-          if (result1.getScore() > result2.getScore()) {
-            return -1
-          }
-          return 0
-        },
-      )
-      return sortedScoredResults
-    }
+  /**
+   * Returns array of revelevant and sorted results
+   * @param results - array of results from all text search adapters
+   * @param rankFn - function that updates results scores
+   * based on more relevance
+   */
+  sortResults(
```

Comment:
This is good refactor, I may have missed it last time I reviewed the proposed changes.

Reply:
thanks! I did some very miscellaneous refactors, none of which should have adjusted correctness of results in any way, just picky stuff!